### DISCUSSION TITLE
Meaning of the MemoryUsed parameter

### DISCUSSION THREAD

**Discussion by arrabito** [DIRACGrid/DIRAC#7757] [Q&A] [answered]
Dear all,
Coming back to the MemoryUsed parameter, as far as I understand from the code it corresponds to the last value of the RSS memory reported by the watchdog subtracted by the initial value:

https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/WorkloadManagementSystem/JobWrapper/Watchdog.py#L804

Is that correct? 

Do I understand correctly that this parameter does not correspond to the peak memory used by the process, neither to the average value? 

Can you also confirm that the watchdog checks occur every 30 minutes?

Thank you,

Luisa

**Comment by fstagni** [comment]
I believe you are correct on everything.

**Comment by arrabito** [comment, marked as answer]
ok thank you, so in this case I think that this parameter is not very useful. 
I undersand indeed that with checks performed every 30 minutes it does not make too much sense to use the collected memory informations neither to compute the average neither to get the maximum.
Anyway it's good to know what this parameter corresponds to, so that we don't really use it to check the memory usage of the jobs.

**Reply by fstagni** [reply]
Maybe https://github.com/HSF/prmon/ could prove useful to you.
For many years I am trying (dreaming?) to find the time to properly add it to DIRAC jobs but never had the time.

**Reply by arrabito** [reply]
Thank you for this link. I will look at it and I take note that it could be useful to add it to DIRAC if we find some time.
